src/pages/data.test.js

```diff
@@ -0,0 +1,427 @@
+        )
+        expect(summaryBoxes[2]).toHaveTextContent(
+            `${
+                importSummaries[0].importCount.ignored +
+                importSummaries[1].importCount.ignored
+            }ignored`
+        )
+
+        const summariesRows = within(submitModal).getAllByTestId(
+            'dhis2-uicore-datatablerow'
+        )
+        expect(summariesRows).toHaveLength(3)
+        expect(summariesRows[1]).toHaveTextContent(
+            anExchange.source.requests[0].name
+        )
+        expect(summariesRows[1]).toHaveTextContent(
+            importSummaries[0].importCount.imported
+        )
+        expect(summariesRows[1]).toHaveTextContent(
+            importSummaries[0].importCount.updated
+        )
+        expect(summariesRows[1]).toHaveTextContent(
+            importSummaries[0].importCount.ignored
+        )
+
+        expect(summariesRows[2]).toHaveTextContent(
+            anExchange.source.requests[1].name
+        )
+        expect(summariesRows[2]).toHaveTextContent(
+            importSummaries[1].importCount.imported
+        )
+        expect(summariesRows[2]).toHaveTextContent(
+            importSummaries[1].importCount.updated
+        )
+        expect(summariesRows[2]).toHaveTextContent(
+            importSummaries[1].importCount.ignored
+        )
+    })
+
+    it('show an error when a exchange data is submitted un-successfully', async () => {
+        const importSummariesWithError = [
+            testImportSummary(),
+            testImportSummary({ status: 'ERROR' }),
+        ]
+        const anExchange = testDataExchange({
+            requests: [testRequest(), testRequest()],
+        })
+
+        const exchanges = [anExchange]
+        const { screen } = setUp(<DataPage />, {
+            aggregateDataExchanges: exchanges,
+            exchangeId: anExchange.id,
+            importSummaryResponse: {
+                importSummaries: importSummariesWithError,
+                status: 'ERROR',
+            },
+        })
+
+        const bottomBar = await screen.findByTestId('bottom-bar')
+
+        const submitButton = within(bottomBar).getByRole('button', {
+            name: 'Submit data',
+        })
+        submitButton.click()
+
+        const submitModal = await screen.findByTestId('dhis2-uicore-card')
+        within(submitModal).getByTestId('confirm-submission-button').click()
+
+        waitFor(() => expect(submitButton).toBeDisabled())
+        const warning = await screen.findByTestId('warning')
+        expect(warning).toBeInTheDocument()
+        expect(warning).toHaveTextContent('There was a problem submitting data')
+    })
+
+    it('should show a progress bar when loading content', async () => {
+        const anExchange = testDataExchange()
+        const exchanges = [anExchange]
+
+        const { screen } = setUp(<DataPage />, {
+            aggregateDataExchanges: exchanges,
+            exchangeId: anExchange.id,
+            loadExchangeDataForever: true,
```

Comment:
just curious: is it possible to test this without the `loadExchangeForever: true`? I would have thought that since you're returning a promise from CustomDataProvider, that it would trigger the loading state before it returns the data?

Reply:
indeed you are right

Reply:
i assumed the same but does not work. do you want to have a look at it together to see if we can figure out why/whynot?